torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -498,10 +566,22 @@ void Reducer::mark_variable_ready(VariableIndex index) {
     TORCH_CHECK(!has_marked_unused_parameters_, common_error);
   }
 
+  // If it was scheduled, wait on allreduce in forward pass that tells us
+  // division factor based on no. of currently participating processes.
+  int divFactor;
+  auto& workHandle = forwardPassWorkHandle_.workHandle_;
+  if (workHandle) {
+    workHandle->wait();
+    at::Tensor& res = forwardPassWorkHandle_.resultTensor_;
+    divFactor = res.item().to<int>();
```

Comment:
I found this in the C++ docs, although not sure if there is a faster way to access the data of a single-element tensor?